homeassistant/components/fritzbox/__init__.py

```diff
@@ -76,7 +77,18 @@ async def async_setup_entry(hass, entry):
         user=entry.data[CONF_USERNAME],
         password=entry.data[CONF_PASSWORD],
     )
-    await hass.async_add_executor_job(fritz.login)
+
+    try:
+        await hass.async_add_executor_job(fritz.login)
+    except LoginError:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_REAUTH},
+                data=entry.data,
```

Comment:
Why not pass the bare `entry` here?

If you put the full entry into the context, you can remove the need to search and iterate for the entry in `_update_entry`.

Reply:
Good point. The docs are showcasing passing in `entry.data`. Being new to Python I was unsure whether it's safe to pass `entry`. I'll change this.